src/ts/universe.ts

```diff
@@ -174,13 +174,21 @@ export class UniversePlayer extends UniverseCircle {
 
     if (!charge) return;
 
+    canvasContext.save();
+
     canvasContext.fillStyle = '#e43';
+    canvasContext.translate(vector.x, vector.y);
+    canvasContext.rotate(orientation - Math.PI / 2);
+
     const chargeWidth = radius;
-    const xPos = vector.x + (radius + chargeWidth / 2) * Math.cos(orientation);
-    const yPos = vector.y + (radius + chargeWidth / 2) * Math.sin(orientation);
-    const angle = orientation - Math.PI / 2;
+    const halfChargeWidth = chargeWidth / 2;
+    const xPos = -halfChargeWidth;
+    const yPos = radius + halfChargeWidth + charge;
+
+    canvasContext.fillRect(xPos, yPos, chargeWidth, -charge);
+    fillPath([xPos, yPos], [xPos + chargeWidth, yPos], [0, yPos + 5]);
```

Comment:
Rather than adding `5`, can you instead make it proportional to the size of the player?